### PR TITLE
perf: support link unfurling sso in message extension

### PR DESCRIPTION
**Pull request by wenytang-ms**
c11213

### CODE REVIEW

File: query-org-user-with-message-extension-sso/teamsBot.ts [hunk 1]
```diff
@@ -104,6 +105,44 @@ export class TeamsBot extends TeamsActivityHandler {
     );
   }
 
+  public async handleTeamsAppBasedLinkQuery(context: TurnContext, query: AppBasedLinkQuery): Promise<any> {
+    return await handleMessageExtensionLinkQueryWithSSO(context,
+      oboAuthConfig,
+      initialLoginEndpoint,
+      ["User.Read.All", "User.Read"],
```
Comment (SLdragon) [inline query-org-user-with-message-extension-sso/teamsBot.ts:112]: Why we need User.Read.All? I remeber that User.Read is enough for profile and photo

Comment (wenytang-ms) [reply query-org-user-with-message-extension-sso/teamsBot.ts:112]: sure, this is sample to tell user how to pass permissions in this api. 
will remove the User.Read.All

File: query-org-user-with-message-extension-sso/README.md [hunk 2]
```diff
@@ -43,6 +44,15 @@ After you have completed the provision and deploy steps in `Deploy the app to Az
 
 ![UpdateScopes](./images/graph-scope-update.png)
 
+### Custom the link unfurling SSO Logic
+- Follow the code in `teamsBot.ts`, custom your query logic in `handleMessageExtensionLinkQueryWithSSO`.
+
+![CustomLogicLink](./images/custom-link-unfurling.png)
+
+- Update the `domains` in `aad.manifest.json` which triggers the preview card.
```
Comment (frankqianms) [inline query-org-user-with-message-extension-sso/README.md:52]: it should be `appPackage/manifest.json`, rather than `aad.manifest.json` here

Comment (wenytang-ms) [reply query-org-user-with-message-extension-sso/README.md]: thanks, updated

File: query-org-user-with-message-extension-sso/README.md [hunk 3]
```diff
@@ -40,7 +40,7 @@ After you have completed the provision and deploy steps in `Deploy the app to Az
 
 ![CustomLogic](./images/custom-query.png)
 
-- Update the scopes in `aad.manifest.json` which used by your Graph Client.
+- Update the scopes in `appPackage/aad.manifest.json` which used by your Graph Client.
```
Comment (frankqianms) [inline query-org-user-with-message-extension-sso/README.md:43]: still not correct. need to change `appPackage/aad.manifest.json` to `aad.manifest.json`

File: query-org-user-with-message-extension-sso/README.md [hunk 4]
```diff
@@ -49,7 +49,7 @@ After you have completed the provision and deploy steps in `Deploy the app to Az
 
 ![CustomLogicLink](./images/custom-link-unfurling.png)
 
-- Update the `domains` in `aad.manifest.json` which triggers the preview card.
+- Update the `domains` in `appPackage/aad.manifest.json` which triggers the preview card.
```
Comment (frankqianms) [inline query-org-user-with-message-extension-sso/README.md:52]: need to change `aad.manifest.json` to `manifest.json`

**Review by frankqianms — APPROVED** [review]
LGTM~ Approving

**Review by SLdragon — APPROVED** [review]

**Pull request merged**